### PR TITLE
feat: Add health checks for route.openshift.io

### PR DESCRIPTION
**Pull request by ishitasequeira**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Signed off by [Ishita Sequeira](@ishitasequeira )

### CODE REVIEW

**Comment by ishitasequeira** [comment]
@jannfis and @jgwest PTAL

File: resource_customizations/route.openshift.io/Route/health.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      health_status.message = "Route is degraded"
+      return health_status
+    else
+      health_status.status = "Progressing"
+      health_status.message = "Route is still getting admitted"
+      return health_status
+    end
+  end
+end
+return health_status
```
Comment (jannfis) [inline resource_customizations/route.openshift.io/Route/health.lua:35]: In the rare case where `obj.status` is `nil`, we'd return an empty `health_status` while we should really return a `Progressing` status.

Can you please initialize `health_status` before returning it?

**Review by jannfis — CHANGES_REQUESTED** [review]
Thanks @ishitasequeira. I have one comment, please see below.

File: resource_customizations/route.openshift.io/Route/health.lua (hunk 1 above)
Comment (ishitasequeira) [reply resource_customizations/route.openshift.io/Route/health.lua:37]: Thank you @jannfis for the review. I have added the Progressing status.

**Review by jannfis — APPROVED** [review]
LGTM, thank you @ishitasequeira

**Pull request merged**